services/instagram_service/src/instagram.py

```diff
@@ -0,0 +1,43 @@
+from instagrapi import Client
+from datetime import datetime
+class InstagramService:
+    def __init__(self, username: str, password: str, proxy: str = None) -> None:
+        self.client = Client(proxy=proxy)
+        try:
+            self.client.login(username, password)
+        except Exception as e:
+            print(f"Error while setting up Instagram client: {e}")
+        self.latest_message_timestamps = {}
+
+    def send_direct_message(self, message: str, sender_username: str) -> None:
+        sender_id = self.client.user_id_from_username(sender_username)
+        sender_id = int(sender_id)
+        recipients = [sender_id]
+        thread_id = self.client.direct_thread(*recipients).id
+        self.client.direct_send(message, thread_id)
+
+    def read_direct_messages(self) -> list:
+        messages = []
+        threads = self.client.direct_threads()
+        for thread in threads:
+            thread_id = thread.id
```

Comment:
ry moving the thread handling logic into separate methods. The code will be easier to read without extensive nesting

Reply:
Method is too long

Reply:
Also, try to logically split big methods with empty lines